public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -122,15 +122,15 @@ function filterRules(props: PanelProps<UnifiedAlertListOptions>, rules: PromRule
       name.toLocaleLowerCase().includes(replacedName.toLocaleLowerCase())
     );
   }
-  if (Object.values(options.stateFilter).some((value) => value)) {
```

Comment:
This was the root of the problem, we only filtered alerts when `any` of the state filters were `true`, meaning that if no states were selected it would show firing alerts with 0 instances (they instances are filtered in `GroupedModeView`)